app/src/org/commcare/sync/ExternalDataUpdateHelper.java

```diff
@@ -10,21 +12,37 @@
 
 public class ExternalDataUpdateHelper {
 
+    public static final String COMMCARE_DATA_UPDATE_ACTION = "org.commcare.dalvik.api.action.data.update";
+
     /**
      * Broadcasts data update to external apps
      * @param c context to send the broadcast with
      * @param updatedCases list of cases updated or created in the update
      */
-    public static void broadcastDataUpdate(Context c, @Nullable ArrayList<String> updatedCases) {
-        Intent i = new Intent("org.commcare.dalvik.api.action.data.update");
+    public static void broadcastDataUpdate(Context c,
+                                           @Nullable ArrayList<String> updatedCases,
+                                           boolean backgroundSync) {
+        Intent i = new Intent(COMMCARE_DATA_UPDATE_ACTION);
         if (updatedCases != null) {
             i.putStringArrayListExtra("cases", updatedCases);
         }
-        c.sendBroadcast(i, "org.commcare.dalvik.provider.cases.read");
+        c.sendBroadcast(i, COMMCARE_CASE_READ_PERMISSION);
 
         // send explicit broadcast to CommCare Reminders App
         i.setComponent(new ComponentName("org.commcare.dalvik.reminders",
                 "org.commcare.dalvik.reminders.CommCareReceiver"));
         c.sendBroadcast(i);
+
+        // Only broadcast to CommCare when it's a background sync
+        if (backgroundSync) {
+            broadcastDataUpdateToCommCare(c);
+        }
```

Comment:
is this necessary ? I would assume not because we are in the same app.

Reply:
@shubham1g5 the broadcast we currently have requires a custom permission that CommCare currently doesn't have, and I don't think app custom permissions are granted automatically, right? And I was trying to avoid having to handle another permission, but happy to rethink this.

Reply:
I am surprised that CC needs to ask for the permissions it itself define. But if that's the case I am happy with the current code.  We should although remove the `if(backgroundSync)` check for this as it doesn't matter to listening activities whether the sync was background or not